### PR TITLE
CI: Add dependency metrics 

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:

This is a draft PR to get a conversation started around how and what we want to track regarding dependencies. Right now this PR takes a rather simple approach and adds the following to the frontend-metrics build task:

- Total bundle folder size
- Total outdated dependencies
- Total number of low vulnerabilities
- Total number of med vulnerabilities
- Total number of high vulnerabilities

Please feel free to weigh with any suggestions or ideas so we can make this 👌!!!

**Special notes for your reviewer**:
I'm not a bash grep regex master. 😕

### CODE REVIEW

**Review by ashharrison90 — APPROVED** [review]
i think this looks good to me 👍 

as part of the wider conversation around dependencies, is there a reason we aren't using something like dependabot to manage our dependency updates? apologies if this is a stupid question.

File: scripts/ci-frontend-metrics.sh [hunk 1]
```diff
@@ -27,4 +39,9 @@ echo "Metrics: {
   \"grafana.ci-code.grafana-ui.mdx\": \"${MDX_COUNT}\",
   \"grafana.ci-code.legacyForms\": \"${LEGACY_FORMS}\",
   \"grafana.ci-code.strictLint.noExplicitAny\": \"${STRICT_LINT_EXPLICIT_ANY}\"
+  \"grafana.ci-code.bundleFolderSize\": \"${TOTAL_BUNDLE}\",
+  \"grafana.ci-code.dependencies.outdated\": \"${OUTDATED_DEPENDENCIES}\",
+  \"grafana.ci-code.dependencies.lowVulnerabilities\": \"${LOW_VULNERABILITIES}\",
+  \"grafana.ci-code.dependencies.mediumVulnerabilities\": \"${MED_VULNERABILITIES}\",
+  \"grafana.ci-code.dependencies.highVulnerabilities\": \"${HIGH_VULNERABILITIES}\",
```
Comment (hugohaggmark) [inline scripts/ci-frontend-metrics.sh:46]: I guess we could also have the following structure:
```suggestion
  \"grafana.ci-code.dependencies.vulnerabilities.low\": \"${LOW_VULNERABILITIES}\",
  \"grafana.ci-code.dependencies.vulnerabilities.medium\": \"${MED_VULNERABILITIES}\",
  \"grafana.ci-code.dependencies.vulnerabilities.high\": \"${HIGH_VULNERABILITIES}\",
```

In this case we can more easily count the total for vulnerabilities, WDYT?

**Review by hugohaggmark — APPROVED** [review]
Regardless of this PR which is so outstanding I also would like to say how impressed I'm with your dependencies work so far Jack 🎉

**Review by mckn — APPROVED** [review]
LGTM! I agree with @hugohaggmark  regarding the naming.

File: scripts/ci-frontend-metrics.sh (hunk 1 above)
Comment (jackw) [reply scripts/ci-frontend-metrics.sh]: Excellent suggestion Hugo, thanks! 🥳

**Comment by jackw** [comment]
> as part of the wider conversation around dependencies, is there a reason we aren't using something like dependabot to manage our dependency updates? apologies if this is a stupid question.

It's a good question @ashharrison90. We do have [dependabot](https://github.com/grafana/grafana/pulls/app%2Fdependabot) but I think it is intentionally limited on bumping dependencies due to the amount of noise it could create. Due to the number of outdated deps I think it would be best to opt for cleaning them up first then have dependabot take over.

File: scripts/ci-frontend-metrics.sh [hunk 2]
```diff
@@ -27,4 +39,9 @@ echo "Metrics: {
   \"grafana.ci-code.grafana-ui.mdx\": \"${MDX_COUNT}\",
   \"grafana.ci-code.legacyForms\": \"${LEGACY_FORMS}\",
   \"grafana.ci-code.strictLint.noExplicitAny\": \"${STRICT_LINT_EXPLICIT_ANY}\"
```
Comment (wbrowne) [inline scripts/ci-frontend-metrics.sh:41]: Missing a sneaky little comma here it looks like 🙈

File: scripts/ci-frontend-metrics.sh [hunk 3]
```diff
@@ -27,4 +39,9 @@ echo "Metrics: {
...
+  \"grafana.ci-code.bundleFolderSize\": \"${TOTAL_BUNDLE}\",
+  \"grafana.ci-code.dependencies.outdated\": \"${OUTDATED_DEPENDENCIES}\",
+  \"grafana.ci-code.dependencies.vulnerabilities.low\": \"${LOW_VULNERABILITIES}\",
+  \"grafana.ci-code.dependencies.vulnerabilities.medium\": \"${MED_VULNERABILITIES}\",
+  \"grafana.ci-code.dependencies.vulnerabilities.high\": \"${HIGH_VULNERABILITIES}\",
```
Comment (wbrowne) [inline scripts/ci-frontend-metrics.sh:46]: Probably need to remove this comma too

**Comment by jackw** [comment]
I'm taking this out of draft now as I've finally got it working and tested it on linux. I believe it will also fix the latest issue we have with noExplicitAny returning 1. This appears to be a result of wrapping the NPM commands in yarn which causes and exit code 2.

**Pull request merged**